storage_service/locations/tests/test_rclone.py

```diff
@@ -0,0 +1,464 @@
+        with pytest.raises(models.StorageException):
+            rclone_space._execute_subprocess(subcommand)
+    else:
+        return_value = rclone_space._execute_subprocess(subcommand)
+        assert return_value == subprocess_communicate_return[0]
+
+
+@pytest.mark.parametrize(
+    "package_is_file, expected_subcommand",
+    [
+        # Package is file, expect "copyto" subcommand
+        (
+            True,
+            [
+                "copyto",
+                "testremote:testcontainer/{}".format(SRC_PATH.lstrip("/")),
+                DEST_PATH,
+            ],
+        ),
+        # Package is directory, expect "copy" subcommand
+        (
+            False,
+            [
+                "copy",
+                "testremote:testcontainer/{}".format(SRC_PATH.lstrip("/")),
+                DEST_PATH + "/",
+            ],
+        ),
+    ],
+)
+def test_rclone_move_to_storage_service(
+    mocker, rclone_space, package_is_file, expected_subcommand
+):
+    exec_subprocess = mocker.patch("locations.models.rclone.RClone._execute_subprocess")
+    mocker.patch(
+        "locations.models.rclone.RClone.remote_prefix",
+        return_value="testremote:",
+        new_callable=mocker.PropertyMock,
+    )
+    mocker.patch("locations.models.rclone.RClone._ensure_container_exists")
+    mocker.patch("common.utils.package_is_file", return_value=package_is_file)
+
+    rclone_space.move_to_storage_service(SRC_PATH, DEST_PATH, rclone_space)
+    exec_subprocess.assert_called_with(expected_subcommand)
+
+
+@pytest.mark.parametrize(
+    "package_is_file, expected_subcommand",
+    [
+        # Package is file, expect "copyto" subcommand
+        (True, ["copyto", "testremote:{}".format(SRC_PATH.lstrip("/")), DEST_PATH]),
+        # Package is directory, expect "copy" subcommand
+        (
+            False,
+            ["copy", "testremote:{}".format(SRC_PATH.lstrip("/")), DEST_PATH + "/"],
+        ),
+    ],
+)
+def test_rclone_move_to_storage_service_no_container(
+    mocker, rclone_space_no_container, package_is_file, expected_subcommand
+):
+    exec_subprocess = mocker.patch("locations.models.rclone.RClone._execute_subprocess")
+    mocker.patch(
+        "locations.models.rclone.RClone.remote_prefix",
+        return_value="testremote:",
+        new_callable=mocker.PropertyMock,
+    )
+    mocker.patch("common.utils.package_is_file", return_value=package_is_file)
+
+    rclone_space_no_container.move_to_storage_service(SRC_PATH, DEST_PATH, rclone_space)
+    exec_subprocess.assert_called_with(expected_subcommand)
+
+
+@pytest.mark.parametrize(
+    "package_is_file, expected_subcommand",
+    [
+        # Package is file, expect "copyto" subcommand
+        (
+            True,
+            [
+                "copyto",
+                SRC_PATH,
+                "testremote:testcontainer/{}".format(DEST_PATH.lstrip("/")),
+            ],
+        ),
+        # Package is directory, expect "copy" subcommand
```

Comment:
Although I appreciate the comments in each case, the use of a single `DEST_PATH` for `Package is file/directory` is a bit hard to follow. For example, `DEST_PATH` seems to be used the same in this test but one is a package and the other a directory?

Maybe you could use different constants for each case like the integration test does:

https://github.com/artefactual/archivematica-storage-service/blob/f75110b8e8cbac92b271d50f1fe2dc4b306137c8/integration/test_integration.py#L29-L34